app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm/reconfigure.rb

```diff
@@ -0,0 +1,112 @@
+module ManageIQ::Providers::IbmPowerHmc::InfraManager::Vm::Reconfigure
+  def reconfigurable?
+    host_hmc_managed
+  end
+
+  def max_total_vcpus
+    # 64 is based on the CEC's CurrentMaximumVirtualProcessorsPerAIXOrLinuxPartition and
+    # CurrentMaximumVirtualProcessorsPerVirtualIOServerPartition settings.
+    # This can be further reduced by the partition's MaximumVirtualProcessors setting.
+    host && try(:processor_share_type) == "dedicated" ? host.cpu_total_cores : 64
+  end
+
+  def max_vcpus
+    max_total_vcpus
+  end
+
+  def max_cpu_cores_per_socket(_total_vcpus = nil)
+    1
+  end
+
+  def max_memory_mb
+    host ? host.hardware.memory_mb : 1_024
+  end
+
+  def build_config_spec(options)
+    $ibm_power_hmc_log.debug("building spec for #{options}")
+
+    lpar = ext_management_system.with_provider_connection { |connection| provider_object(connection) }
```

Comment:
Could you just use `provider_object` directly here since it will connect automatically?
```suggestion
    lpar = provider_object
```

Reply:
Unfortunately, no, because it wouldn't log off and would leave a dangling session on the HMC.

Reply:
Okay I wonder if we should (not for this PR) get rid of the raw `provider_object` method (or at least the optional `connection` with the `connection ||=`) and just use the `with_provider_object` which can release the connection as part of the block.

https://github.com/ManageIQ/manageiq-providers-ibm_power_hmc/blob/master/app/models/manageiq/providers/ibm_power_hmc/infra_manager/lpar.rb#L6-L9